src/sql/workbench/services/connection/browser/connectionDialogService.ts

```diff
@@ -487,7 +501,7 @@ export class ConnectionDialogService implements IConnectionDialogService {
 			});
 			this._connectionDialog.onShowUiComponent((input) => this.handleShowUiComponent(input));
 			this._connectionDialog.onInitDialog(() => this.handleInitDialog());
-			this._connectionDialog.onFillinConnectionInputs((input) => this.handleFillInConnectionInputs(input as IConnectionProfile));
+			this._connectionDialog.onFillinConnectionInputs(async (input) => this.handleFillInConnectionInputs(input as IConnectionProfile));
```

Comment:
nit: This doesn't need to be async if you aren't awaiting anything.

Reply:
I thought the rule was that you need to declare something as async if it's calling another async function, but awaiting it was optional if you wanted the promise to bubble up the call hierarchy.

Reply:
Async does nothing if there isn't an await and it isn't returning anything. And even then you only need async if the return value isn't a promise already itself - in this case for example you could just return the call to handleFillInConnectionInputs directly and it would then properly bubble up the promise to the caller.

So in this case async isn't going to do anything useful - and you also aren't even passing up the promise. (but events don't do anything with rejected promises anyways so that isn't something we generally put much effort into doing anything about - better to do like you do and have the called function handle any errors itself)